test/src/main/java/org/openjdk/skara/test/TestPullRequest.java

```diff
@@ -242,17 +243,24 @@ public URI headUrl() {
     @Override
     public Diff diff() {
         try {
-            var targetHash = targetRepository.branchHash(targetRef());
             var targetLocalRepository = targetRepository.localRepository();
             var sourceLocalRepository = sourceRepository.localRepository();
-            if (targetLocalRepository.root().equals(sourceLocalRepository.root())) {
-                // same target and source repo, must contain both commits
-                return targetLocalRepository.diff(targetHash, headHash());
-            } else {
-                var uri = URI.create("file://" + sourceLocalRepository.root().toString());
-                var fetchHead = targetLocalRepository.fetch(uri, sourceRef);
-                return targetLocalRepository.diff(targetHash, fetchHead);
+            var sourceHash = headHash();
+            if (!targetLocalRepository.root().equals(sourceLocalRepository.root())) {
+                // The target and source repo are not same, fetch the source branch
+                var sourceUri = URI.create("file://" + sourceLocalRepository.root().toString());
+                sourceHash = targetLocalRepository.fetch(sourceUri, sourceRef);
             }
+            // Find the base hash of the source and target branches.
+            var baseHash = sourceHash;
+            var targetBranch = new Branch(targetRef());
+            while (!"0".repeat(40).equals(baseHash.hex())) {
```

Comment:
So if the common commit is more than 40 commits away, return "no diff"? :-)

This is only used in testing, right? If so, this you could get away with it, but I think a comment describing the limitations would be needed.

Reply:
I think you are misunderstanding the code. `"0".repeat(40)` just generates a string with 40 zeros, basically a null hash.

Looping to find the common ancestor is one way of doing it, but I would recommend checking out `git merge-base`. There is a method for it already on `GitRepository`.

Reply:
I see. Yes, I misunderstood the code as looping backwards for a maximum of 40. Don't know how I got to that conclusion. My brain is apparently still partially stuck in vacation mode. I don't think I'm used to methods being called on a string literal.

Reply:
> Looping to find the common ancestor is one way of doing it, but I would recommend checking out `git merge-base`. There is a method for it already on `GitRepository`.

Thanks for reminding me. I change the code to use the method `mergeBase` just now.